### ISSUE TITLE
[BUG] FT.AGGREGATE Inconsistent Multi-Sorting in Redis 6.0.20 with RedisSearch v2.6.12

### ISSUE THREAD

**Issue by shrihari-tc** [RediSearch/RediSearch#4711]
Hi team.
We are experiencing an issue with the FT.AGGREGATE command in Redis 6.0.20, specifically with RedisSearch v2.6.12 and RedisJSON v2.4.7. When using the FT.AGGREGATE command with multiple sort options, only the first sort option is being considered, and the rest are being ignored.

_**NOTE**_ : The sort results are not consistent every time I run the command. Sometimes the results come as expected, while other times they ignore the additional sort options.

**To Reproduce**
1. Use Redis 6.0.20 with RedisSearch v2.6.12 and RedisJSON v2.4.7.
2. Set the JSON data for two users:
```
JSON.SET employee:user1 $ '{"firstName":"Aron", "age": 12, "lastName":"snow"}'
JSON.SET employee:user2 $ '{"firstName":"John", "age": 10, "lastName":"water"}'
JSON.SET employee:user3 $ '{"firstName":"John", "lastName":"yeno"}'
JSON.SET employee:user4 $ '{"firstName":"John", "lastName":"zeno"}'
```
3. Create an index on the firstName and lastName fields:
```
FT.CREATE idx_employee ON JSON PREFIX 1 employee: SCHEMA $.firstName AS firstName TEXT SORTABLE $.lastName AS lastName TEXT SORTABLE $.age AS age NUMERIC SORTABLE
```
4. Run the FT.AGGREGATE command with multiple sort options:
```
FT.AGGREGATE idx_employee "*" LOAD 3 age firstName lastName SORTBY 4 @age DESC @lastName DESC
```
**Expected behavior**
The results should be sorted based on all specified sort options:
```
[
    {
        "firstName": "Aron",
        "age": "12",
        "lastName": "snow"
    },
    {
        "firstName": "John",
        "age": "10",
        "lastName": "water"
    },
    {
        "firstName": "John",
        "lastName": "zeno"
    },
    {
        "firstName": "john",
        "lastName": "yeno"
    }
]

```

**Actual Behavior**
```
[
    {
        "firstName": "Aron",
        "age": 12,
        "lastName": "snow"
    },
    {
        "firstName": "John",
        "age": 10,
        "lastName": "water"
    },
    {
        "firstName": "john",
        "lastName": "yeno"
    },
    {
        "firstName": "John",
        "lastName": "zeno"
    }
]
```

**Environment:**
- Redis Version: 6.0.20
- RedisSearch Version: 2.6.12
- RedisJSON Version: 2.4.7
- Deployment: Azure Cache for Redis

**Additional context**
We are currently using Azure Cache for Redis and are not in a position to upgrade to a newer version of Redis or its modules.

Any assistance or workaround to resolve this issue would be greatly appreciated.

**Comment by shrihari-tc** [comment]
<img width="907" alt="ss-with-table" src="https://github.com/RediSearch/RediSearch/assets/120747379/7c01eca8-0e2e-4014-ac00-9fb1825ad648">

Here is the screenshot of the inconsistent result.

**Comment by GuyAv46** [comment]
OK, so I was able to track the root cause of the issue here. The issue is with `cmpByField()`

On [2.8](https://github.com/RediSearch/RediSearch/blob/v2.8.3/src/result_processor.c#L459-L498) ([this line](https://github.com/RediSearch/RediSearch/blob/v2.8.3/src/result_processor.c#L481), to be specific), we fixed the comparison logic of two rows with the same primary sorting key missing.

In my script, since I added `user3` before `user4`, its internal ID was smaller and it was sorted **after** `user4` due to the `DESC` flag on the current (missing) sorting key.
Internal DocID is used for sorting as a last resort, as a best effort to have consistent results, but when you have a cluster of redis instances, different docs in different instances might have the same ID and therefore it won't help with a consistent result.

Unfortunately, this is not fixed on 2.6 yet, so your version suffers from this bug.
As I see it, there is no perfect workaround. If some documents are missing the primary key, it's hard to predict their relative order (compared to other docs with a valid value, they will be "worse" whether we sort them in ascending or descending order).

You may want to consider a default value for docs with missing fields (in your case, you can set it on your application that if age is not specified, you set it to 0). This is a bit limiting, but if you always sort in the same way (`ASC`/`DESC`) it might be useful.

In any case, I'll open a PR to fix this bug on 2.6. Let's hope the next Azure update will include this fix
